# Problema 1

Las matrices $A$ y $B$ representan las ventas en dos años, en millones de dólares, de los tres productos de una empresa en las cuatro regiones en que está disponible. Es decir, $a_{ij}$ y $b_{ij}$ representan el monto de las ventas del producto $i$ en la región $j$ el primer y segundo año, respectivamente:

<br>

\begin{split}
A = \begin{pmatrix}
 2,6 &  4,8 &  1,8 & 0,9 \\
 3,2 &  4,4 &  2,5 & 2,8 \\
 2,4 &  3,6 &  3,8 & 2,5 \\
\end{pmatrix} \text{ } B= \begin{pmatrix}
 3,6 &  2,5 &  3,0 & 2,5 \\
 4,5 &  5,0 &  3,5 & 3,8 \\
 2,9 &  3,0 &  4,6 & 4,0 \\
\end{pmatrix}
\end{split}

<br>

1. Indique el orden de cada matriz.
2. Calcule la matriz $S = A + B$.
3. Calcule la matriz $D = B - A$.
4. Indique el valor e interprete los elementos $s_{24}$ y $d_{24}$.
5. Para el tercer año de funcionamiento, los tres productos ahora están disponibles en seis regiones. Durante este año, las ventas de esta empresa estarán registradas en una matriz $E$. ¿Es posible determinar la matriz $T = A + B + E$? Justifique su respuesta.

In [30]:
import numpy as np
# Ejemplo de interpretacion
# Si se habla de a_23 hablaremos que el producto 2 de la region 3, la venta en los ultimos 2 años es de 2.5 millones de dolares
A = np.array([
     [2.6, 4.8, 1.8, 0.9],
     [3.2, 4.4, 2.5, 2.8],
     [2.4, 3.6, 3.8, 2.5]
    ])

B = np.array([
     [3.6, 2.5, 3.0, 2.5], 
     [4.5, 5.0, 3.5, 3.8],
     [2.9, 3.0, 4.6, 4.0]
    ])

def calculoMatriz(A,B):
    temp = list()
    S = list()
    for A in A:
        return A 



print(f'El orden de la Matriz A es : {len(A)} x {len(A[0])}')
print(f'El orden de la Matriz B es : {len(B)} x {len(B[0])}')

S = A + B 
D = B - A

print('Matriz S es:\n')
for i in range(len(S)):
    for j in S[i]:
        print(f'{j:.1f}', end='   ')
    print()

    #for j in range(len(A[0])):
    #    print(f'{A[j]}')
print()
print(f'Matris D es:\n')
for i in range(len(D)):
    for j in D[i]:
        print(f'{j:.1f}', end='   ')
    print()

print()
print(f'Valor s[2][4]: {S[1][3]}')

print('Sumando ambos años el producto 2 de la region 4 vendio un total de 6.6 Millones')

print(f'Valor d[2][4]: {D[1][3]}')

print('El producto 2 de la region 4 vendio 1 millon mas que lo especificado en el año A')
# Para transponer matrices, se intercambia Fila por Columna
# Las matrices se leen de la forma Fila, Columna
print('No es posible hacer el calculo de una matriz 3x6 con 3x4, ilegalisimo')



El orden de la Matriz A es : 3 x 4
El orden de la Matriz B es : 3 x 4
Matriz S es:

6.2   7.3   4.8   3.4   
7.7   9.4   6.0   6.6   
5.3   6.6   8.4   6.5   

Matris D es:

1.0   -2.3   1.2   1.6   
1.3   0.6   1.0   1.0   
0.5   -0.6   0.8   1.5   

Valor s[2][4]: 6.6
Sumando ambos años el producto 2 de la region 4 vendio un total de 6.6 Millones
Valor d[2][4]: 1.0
El producto 2 de la region 4 vendio 1 millon mas que lo especificado en el año A


# Problema 2

Una fábrica con sucursales en Japón y China, produce tres tipos de componentes $j$, para cinco modelos de computador $i$. Los costos de producción mensual, en miles de dólares, de ambas sucursales se representan en las matrices $J$ y $C$, respectivamente:

<br>

\begin{split}
J = \begin{pmatrix}
 100 &  200 &  150  \\
 120 &  210 &  170  \\
 112 &  200 &  131  \\
 124 &  310 &  170  \\
 136 &  156 &  140  \\
\end{pmatrix} \text{ } C = \begin{pmatrix}
 110 &  190 &  160  \\
 130 &  220 &  180  \\
 146 &  190 &  141  \\
 143 &  312 &  163  \\
 137 &  116 &  156  \\
\end{pmatrix}
\end{split}

<br>

1. ¿Cuál es el orden de las matrices?
2. Inteprete el elemento $j_{21}$ y $c_{12}$ en el contexto del problema.
3. Utilizando *Python*, calcule el costo total de producción para cada modelo, considerando ambas sucursales.
4. Un trabajador indica que, en la sucursal japonesa, el costo de producción del producto $3$, modelo $4$ de computador, es más alto que en la sucursal china. ¿Tiene razón? Justifique apropiadamente.

In [53]:
import numpy as np 

J = np.array([[100,200,150],
              [120,210,170],
              [112,200,131],
              [124,310,170],
              [136,156,140]])

C = np.array([[110,190,160],
              [130,220,180],
              [146,190,141],
              [143,312,163],
              [137,116,156]])

print(f"El orden de ambas matrices es {len(C)} x {len(C[0])}")
print(f'El elemento J[2][1] indica que los costos de produccion mensuales de la fabrica de japon seran de {J[1][0]} mil dolares para el componente 1, modelo 2 de computador')
print(f'El elemento C[1][2] indica que los costos de produccion mensuales de la fabrica de china seran de {C[0][1]} mil dolares para el componente 2, modelo 1 de computador')

S = J + C
modelos = list()
print(S)

for i in (S):
    suma = 0
    for j in i:
        suma += j;
    modelos.append(suma)

for i, modelo in enumerate (modelos,1):
    print(f'Costo de produccion total para el modelo {i} {modelo}') 

El orden de ambas matrices es 5 x 3
El elemento J[2][1] indica que los costos de produccion mensuales de la fabrica de japon seran de 120 mil dolares para el componente 1, modelo 2 de computador
El elemento C[1][2] indica que los costos de produccion mensuales de la fabrica de china seran de 190 mil dolares para el componente 2, modelo 1 de computador
[[210 390 310]
 [250 430 350]
 [258 390 272]
 [267 622 333]
 [273 272 296]]
Costo de produccion total para el modelo 1 910
Costo de produccion total para el modelo 2 1030
Costo de produccion total para el modelo 3 920
Costo de produccion total para el modelo 4 1222
Costo de produccion total para el modelo 5 841


# Problema 3

La empresa Beta tiene cinco plantas en Chile, y en cada una de ellas fabrica siete productos. El número de unidades del producto $i$ fabricadas en la planta $j$ en una semana está representado en la matriz $C$ en el elemento $c_{ij}$.

<br>

\begin{split}
C = \begin{pmatrix}
 120 &  200 &  155 &  125  &  230 \\
 155 &  130 &  110 &  160  &  180 \\
 140 &  205 &  150 &  125  &  155 \\
 130 &  120 &  230 &  225  &  125 \\
 135 &  70  &  190 &  175  &  95  \\
 110 &  265 &  180 &  230  &  180 \\
 185 &  85  &  120 &  135  &  150 \\
\end{pmatrix}
\end{split}

<br>

La empresa se expande, creando cinco plantas en Ecuador que producen los mismos siete productos. La producción semanal de esta filial estará representada por una matriz $E$.

1. Si se sabe que los volúmenes de producción semanal de la filial en Ecuador son un $20\%$ menor que en Chile, calcule la matriz $E$ utilizando *Python*.
2. Si representamos la producción semanal total de la empresa Beta, considerando ambas filiales, por la matriz $T$, escriba la expresión que calcula cada elemento de la matriz $T$ usando notación matricial.
3. Calcule la matriz $T$ utilizando *Python*.
4. Interprete los elementos $c_{34}$ y $t_{72}$ en el contexto del problema.

In [75]:
import numpy as np

C = np.array([
    [120, 200, 155, 125, 230],
    [155, 130, 110, 160, 180],
    [140, 205, 150, 125, 155],
    [130, 120, 230, 225, 125],
    [135, 70, 190, 175, 95],
    [110, 265, 180, 230, 180],
    [185, 85, 120, 135, 150]
    ])

# Columnas = Plantas en Ecuador
# Filas = Mismos productos

E = C * 0.8

print('Matriz E')
for i in E:
    for j in i:
        print(f'{j:.1f}',end='   ')
    print()

T = E + C

print()
print('Matriz T')
for i in T:
    for j in i:
        print(f'{j:.1f}',end='   ')
    print()

print(f'C[3][4] El producto 3 de la planta 4 de Chile produce {C[2][3]} unidades semanalmente')
print(f'T[7][2] La produccion total del producto 7 de la planta 2 de Chile y Ecuador es de {T[6][1]:.0f} unidades semanales')

Matriz E
96.0   160.0   124.0   100.0   184.0   
124.0   104.0   88.0   128.0   144.0   
112.0   164.0   120.0   100.0   124.0   
104.0   96.0   184.0   180.0   100.0   
108.0   56.0   152.0   140.0   76.0   
88.0   212.0   144.0   184.0   144.0   
148.0   68.0   96.0   108.0   120.0   

Matriz T
216.0   360.0   279.0   225.0   414.0   
279.0   234.0   198.0   288.0   324.0   
252.0   369.0   270.0   225.0   279.0   
234.0   216.0   414.0   405.0   225.0   
243.0   126.0   342.0   315.0   171.0   
198.0   477.0   324.0   414.0   324.0   
333.0   153.0   216.0   243.0   270.0   
C[3][4] El producto 3 de la planta 4 de Chile produce 125 unidades semanalmente
T[7][2] La produccion total del producto 7 de la planta 2 de Chile y Ecuador es de 153 unidades semanales


# Problema 4

Una empresa que fabrica zapatos tiene dos plantas, una en Valparaíso y la otra en Temuco. La empresa produce zapatos de color negro, blanco y café, tanto para niños, damas como caballeros. La capacidad actual de producción mensual (en miles de pares) para Valparaíso y Temuco, están dadas por las matrices $V$ y $T$, respectivamente. Para ambas matrices, según el orden en que se mencionan, las filas corresponden al color del calzado y las columnas al tipo de persona que utilizará el calzado.

<br>

\begin{split}
V = \begin{pmatrix}
 30 &  34 &  20  \\
 45 &  20 &  16  \\
 14 &  26 &  25  \\
\end{pmatrix} \text{ } T = \begin{pmatrix}
 35 &  30 &  26  \\
 52 &  25 &  18  \\
 23 &  24 &  32  \\
\end{pmatrix}
\end{split}

<br>

1. Determine la producción mensual total de ambas plantas, representándola en una matriz $A$. Utilice *Python* para resolver.
2. Interprete el elemento $a_{21}$ en el contexto del problema.
3. Si para el próximo año se estima que la producción en Valparaíso aumenta en un $30\%$, mientras que la de Temuco se reduce en un $10\%$, represente la nueva producción total en la matriz $B$.
4. Interprete el elemento $b_{13}$ en el contexto del problema.

In [86]:
# Filas: Color del calzado
# Columnas: Tipo de personas

import numpy as np

V = np.array([
    [30, 34, 20],
    [45, 20, 16],
    [14, 26, 25]
])

T = np.array([
    [35, 30, 26],
    [52, 25, 18],
    [23, 24, 32]
])

A = V + T

for i in A:
    for j in i:
        print(round(j,1),end='  ')
    print()

print(f'La capacidad de produccion mensual en A[2][1] para los pares blancos de zapatillas para niños de ambas fabricas juntas es de {A[1][0]} mil pares')

Va = V * 1.3
Te = T * 0.9

B = Va + Te

print(f'La capacidad de produccion mensual en B[1][3] para los pares negros de zapatillas para caballeros de ambas fabricas juntas para el proximo año es de {B[0][2]:.1f} mil pares')

65  64  46  
97  45  34  
37  50  57  
La capacidad de produccion mensual en A[2][1] para los pares blancos de zapatillas para niños de ambas fabricas juntas es de 97 mil pares
La capacidad de produccion mensual en B[1][3] para los pares negros de zapatillas para caballeros de ambas fabricas juntas para el proximo año es de 49.4 mil pares


# Problema 5

Genere una matriz $M$ de orden $8 \times 9$ donde cada término de la matriz está definido por:

$$m_{ij} = 8i + 12j$$

y calcule su transpuesta.

In [ ]:
import numpy as np

M = np.array([])
for i in range(1,13):
    temp = list()  
    for j in range(1,9):
        if len(M):
            temp.append(8*i + 12*j)
        else:
            temp.append(temp)


# Problema 6

Genere dos matrices:

*   Una matriz $A$ de orden $5 \times 7$, con $a_{ij} = i \cdot j.$
*   Una matriz $B$ de orden $7 \times 5$, con $b_{ij} = 15i - 2j.$

y calcule $C = A^{T} + \dfrac{1}{2}B$.



# Problema 7

Una red de sensores recoge datos de temperatura ($°C$), humedad (porcentaje) y presión (hPa) en cinco días distintos (de lunes a viernes). Los datos medidos se muestran en la matriz $D$:

<br>

\begin{equation*}
\begin{matrix}
& \begin{matrix}
 & Lu  \ \ &Ma \ \ &Mi \ \ & Ju \ \ &Vi
\end{matrix} \\
&D = \begin{pmatrix}
20    &  28    &  22    &  26     &  24   \\
52    &  57    &  56    &  54     &  58   \\
1.066 &  1.068 &  1.066 &  1.064  &  1.068 \\
\end{pmatrix} & \ \begin{matrix}
Cº\\
\% \\
hPa
\end{matrix}\\
\end{matrix}
\end{equation*}

<br>

Se sabe que por un error de calibración de los sensores, las medidas registradas deberían ser un $5\%$ menores que las registradas.

1. Calcule en una matriz $S$ los valores correctos de temperatura, humedad y presión. Utilice *Python* para calcular.
2. Obtenga la matriz $S^{T}$.
3. Interprete los elementos $s_{24}$ y $s_{31}^{T}$.


# Problema 8

Una empresa tecnológica vende software y hardware en tres productos: $P_{1}$, $P_{2}$ y $P_{3}$, en cuatro zonas de la capital: norte, sur, este y oeste. Las ventas del mes, en millones de pesos, están registradas en las matrices $S$ y $H$, para software y hardware respectivamente.

<br>

\begin{split}
S = \begin{pmatrix}
 260 &  480 &  180  &  90  \\
 320 &  440 &  250  &  280 \\
 240 &  360 &  380  &  250  \\
\end{pmatrix} \text{ } H = \begin{pmatrix}
 360 &  250 &  300  &  250  \\
 450 &  500 &  350  &  380 \\
 290 &  300 &  460  &  400  \\
\end{pmatrix}
\end{split}

<br>

1. Calcule en la matriz $T$, del total de ventas de la empresa tecnológica.
2. Un trabajador indica que este mes, la sucursal de la zona sur recaudó más por la venta hardware que software, ¿es correcta su afirmación?
3. Debido al black friday se espera que el siguiente mes, las ventas de software aumenten un $7\%$ y las de hardware aumenten un $3\%$ respecto al mes actual. Encuentra en total de ventas esperadas para el siguiente mes y almacénalo en la matriz $M$. Utiliza *Python* para realizar los cálculos.
4. Interpreta los elementos $t_{24}$ y $m_{31}$ en el contexto del problema.
5. Calcula la matriz $M^{T}$ e interpreta el elemento $m_{41}^{T}$ en el contexto del problema.
